shared/fs/common/path-item-action.js

```diff
@@ -0,0 +1,129 @@
+import {globalStyles, globalColors, globalMargins, isMobile, platformStyles} from '../../styles'
+import {Box, ClickableBox, Icon, Text} from '../../common-adapters'
+import FloatingMenu, {
+  FloatingMenuParentHOC,
+  type FloatingMenuParentProps,
+} from '../../common-adapters/floating-menu'
+import {type MenuItem} from '../../common-adapters/popup-menu'
+import PathItemIcon from '../common/path-item-icon'
+import PathItemInfo from '../common/path-item-info'
+import StaticBreadcrumb from '../common/static-breadcrumb'
+import {memoize} from 'lodash'
+
+type Props = {
+  name: string,
+  size: number,
+  type: Types.PathType,
+  lastModifiedTimestamp: number,
+  lastWriter: string,
+  childrenFolders: number,
+  childrenFiles: number,
+  itemStyles: Types.ItemStyles,
+  actionIconClassName?: string,
+  actionIconFontSize?: number,
+  pathElements: Array<string>,
+  menuItems: Array<MenuItem>,
+}
+
+const PathItemAction = (props: Props & FloatingMenuParentProps) => (
+  <Box>
+    <ClickableBox onClick={props.toggleShowingMenu} ref={props.setAttachmentRef}>
+      <Icon
+        type="iconfont-ellipsis"
+        color={globalColors.black_40}
+        style={actionIconStyle}
+        fontSize={props.actionIconFontSize}
+        className={props.actionIconClassName}
+      />
+    </ClickableBox>
+    <FloatingMenu
+      containerStyle={stylesFloatingContainer}
+      attachTo={props.attachmentRef}
+      visible={props.showingMenu}
+      onHidden={props.toggleShowingMenu}
+      position="bottom right"
+      closeOnSelect={true}
+      header={{
+        title: 'unused',
+        view: (
+          <Box style={stylesHeader}>
+            <PathItemIcon spec={props.itemStyles.iconSpec} style={pathItemIconStyle} />
+            <StaticBreadcrumb pathElements={props.pathElements} />
+            <Box style={stylesNameTextBox}>
+              <Text type="BodySmallSemibold" style={stylesNameText(props.itemStyles.textColor)}>
+                {props.name}
+              </Text>
+            </Box>
+            {props.type === 'file' && (
+              <Text type="BodySmall">{Constants.humanReadableFileSize(props.size)}</Text>
+            )}
+            {props.type === 'folder' && (
+              <Text type="BodySmall">
+                {props.childrenFolders
+                  ? `${props.childrenFolders} Folder${props.childrenFolders > 1 ? 's' : ''}${
+                      props.childrenFiles ? ', ' : ''
+                    }`
+                  : undefined}
+                {props.childrenFiles
+                  ? `${props.childrenFiles} File${props.childrenFiles > 1 ? 's' : ''}`
+                  : undefined}
+              </Text>
+            )}
+            <PathItemInfo
+              lastModifiedTimestamp={props.lastModifiedTimestamp}
+              lastWriter={props.lastWriter}
+              wrap={true}
+            />
+          </Box>
+        ),
+      }}
+      items={props.menuItems.map(a => a /* make flow happy */)}
```

Comment:
cause some can be null? you can do
`props.menuItems.filter(Boolean)`

Reply:
No I think it's because `menuItems` here is a different array, despite it's actually more restrictive:

```
$ /Users/songgao/gopath/src/github.com/keybase/client/shared/node_modules/.bin/flow
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ fs/common/path-item-action.js:82:14

Cannot create FloatingMenu element because:
 • inexact string literal Divider [1] is incompatible with exact MenuItem [2] in array element of property items.
 • inexact null [3] is incompatible with exact MenuItem [2] in array element of property items.

        fs/common/path-item-action.js
    [2] 29│   menuItems: Array<MenuItem>,
          :
        68│         className={props.actionIconClassName}
        69│       />
        70│     </ClickableBox>
        71│     <FloatingMenu
        72│       containerStyle={stylesFloatingContainer}
        73│       attachTo={props.attachmentRef}
        74│       visible={props.showingMenu}
        75│       onHidden={props.toggleShowingMenu}
        76│       position="bottom right"
        77│       closeOnSelect={true}
        78│       header={{
        79│         title: 'unused',
        80│         view: <PathItemActionHeader {...props} />,
        81│       }}
        82│       items={props.menuItems}
        83│     />
        84│   </Box>
        85│ )
        86│

        common-adapters/floating-menu.js
 [1][3] 12│   items: Array<MenuItem | 'Divider' | null>,
```

Reply:
I'll just change the `menuItems` type here.